spec/defines/router_db_spec.rb

```diff
@@ -56,4 +56,19 @@
                .with_mode(0123)
     }
   end
+
+  context 'managing remote urls' do
+    let(:params) {
+      { :vcs_remote_urls => {
+          'group1' => 'http://my-server/my-path.git',
+        },
+      }
+    }
+
+    it {
```

Comment:
also should test the two file resources added

Reply:
The existing test asserts that we do in fact behave correctly if asked to manage remote urls (though I should probably add a negative test).

That being the case, whilst I'm happy to add the tests you request, what benefit would they add over the existing test?

Reply:
Besides the exec, which you test for, you also added two file resources  `"post-commit hook for ${name}"` and `"rancid default git remote ${name}"` which should be tested in that same block and are not being tested elsewhere.

Reply:
Given that all three resource are enclosed in the same conditional block, there is no way that any of the three can be included in the catalogue without _all_ of them being included.  Therefore, there is no benefit including tests for all three, because they will either pass or fail together.  I included a test for the exec purely to test that the condition is working correctly.  If we add tests for the other two resources, then surely we're simply repeating the manifest, but not really adding any value?

Reply:
You do not need to include all of the attributes though there should be basic, this resource should exist. If the coverage bits were enabled you would get a report at the end showing untested resources. While this might seem like extra work now, as the code changes over time, this ensures proper testing.

```ruby
 it { should contain_file('asdf') }
```